bins/ggipcd/src/services/authorizationagent/token_validator.c

```diff
@@ -28,33 +29,52 @@ GglError ggl_handle_token_validation(
         )) {
         GGL_LOGE(
             "Component %.*s does not have access to token verification IPC "
-            "command",
+            "command.",
             (int) info->component.len,
             info->component.data
         );
+
+        *ipc_error = (GglIpcError
+        ) { .error_code = GGL_IPC_ERR_UNAUTHORIZED_ERROR,
+            .message = GGL_STR(
+                "Component does not have access to token verification IPC "
+                "command."
+            ) };
+
+        return GGL_ERR_FAILURE;
     }
+
     GglObject *svcuid_obj;
     GglError ret = ggl_map_validate(
         args,
         GGL_MAP_SCHEMA({ GGL_STR("token"), true, GGL_TYPE_BUF, &svcuid_obj }, )
     );
     if (ret != GGL_ERR_OK) {
         GGL_LOGE("Received invalid paramters.");
+        *ipc_error = (GglIpcError) { .error_code = GGL_IPC_ERR_SERVICE_ERROR,
+                                     .message
+                                     = GGL_STR("Failed to validate the map.") };
         return GGL_ERR_INVALID;
     }
 
-    if (ipc_svcuid_exists(svcuid_obj->buf) == GGL_ERR_OK) {
+    if (ipc_svcuid_exists(svcuid_obj->buf) != GGL_ERR_OK) {
+        *ipc_error = (GglIpcError
+        ) { .error_code = GGL_IPC_ERR_INVALID_TOKEN_ERROR,
+            .message = GGL_STR(
+                "Invalid token used by stream manager when trying to authorize."
+            ) };
         return ggl_ipc_response_send(
             handle,
             stream_id,
             GGL_STR("aws.greengrass#ValidateAuthorizationTokenResponse"),
-            GGL_OBJ_MAP(GGL_MAP({ GGL_STR("isValid"), GGL_OBJ_BOOL(true) }))
+            GGL_OBJ_MAP(GGL_MAP({ GGL_STR("isValid"), GGL_OBJ_BOOL(false) }))
```

Comment:
Nice fix

Reply:
This is incorrect fix? looks like type confusion

Reply:
I actually flipped the sign too :)